apolloschurchapp/src/tabs/index.js

```diff
@@ -1,27 +1,103 @@
 import React from 'react';
-import { Platform } from 'react-native';
+import { Image } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
 import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
-import { withTheme } from '@apollosproject/ui-kit';
+import { createNativeStackNavigator } from 'react-native-screens/native-stack';
+import { styled } from '@apollosproject/ui-kit';
 
+import { SearchButton } from '../ui/Search';
+import Tab from './Tab';
 import Connect from './connect';
-import Home from './home';
-import Discover from './discover';
 import tabBarIcon from './tabBarIcon';
 
+const HeaderLogo = styled(({ theme }) => ({
+  height: theme.sizing.baseUnit,
+  resizeMode: 'contain',
+}))(Image);
+
+const HeaderCenter = () => <HeaderLogo source={require('./wordmark.png')} />;
+const HeaderRight = () => {
+  const navigation = useNavigation();
+  return <SearchButton onPress={() => navigation.navigate('Search')} />;
+};
+
+// we nest stack inside of tabs so we can use all the fancy native header features
+const HomeStack = createNativeStackNavigator();
+const Home = () => <Tab tab={'HOME'} />;
+const HomeTab = () => (
+  <HomeStack.Navigator
+    screenOptions={{
+      headerHideShadow: true,
+      headerCenter: HeaderCenter,
+      headerRight: HeaderRight,
+    }}
+  >
+    <HomeStack.Screen name="Home" component={Home} />
+  </HomeStack.Navigator>
+);
+
+const ReadStack = createNativeStackNavigator();
+const Read = () => <Tab tab={'READ'} />;
+const ReadTab = () => (
+  <ReadStack.Navigator
+    screenOptions={{
+      headerHideShadow: true,
+      headerLargeTitle: true,
+    }}
+  >
+    <ReadStack.Screen name="Read" component={Read} />
+  </ReadStack.Navigator>
+);
+
+const WatchStack = createNativeStackNavigator();
+const Watch = () => <Tab tab={'WATCH'} />;
+const WatchTab = () => (
+  <WatchStack.Navigator
+    screenOptions={{
+      headerHideShadow: true,
+      headerLargeTitle: true,
+    }}
+  >
+    <WatchStack.Screen name="Watch" component={Watch} />
+  </WatchStack.Navigator>
+);
+
+const PrayStack = createNativeStackNavigator();
```

Comment:
Would like @vinnyjth thoughts on structuring all of these inside of one file instead of separated. I don't have strong opinion